audio_program_generator/apg.py

```diff
@@ -118,14 +120,15 @@ def gen_speech(self):
                 continue
 
             # Cache genearted gTTTS snippets and reuse if already present
-            Path.mkdir(Path.cwd() / ".cache") if not Path(
-                Path.cwd() / ".cache"
-            ).exists() else None
-
-            file = Path.cwd() / ".cache" / (phrase + ".mp3")
-            if not Path(file).exists():
-                speech = gTTS(phrase)
-                speech.save(file)
+            if self.cache:
+                Path.mkdir(Path.cwd() / ".cache") if not Path(
+                    Path.cwd() / ".cache"
+                ).exists() else None
+
+                file = Path.cwd() / ".cache" / (phrase + ".mp3")
+                if not Path(file).exists():
+                    speech = gTTS(phrase)
+                    speech.save(file)
```

Comment:
This code was reworked in 1.6.0 to implement in-memory "file" handling, and caching was removed. `main` in 1.6.0 should be able to just be taken as-is.

Reply:
ok cool, yep I will